app/views/articles/index.html.haml

```diff
@@ -1,18 +1,29 @@
 %h1 List of articles
+= render 'search', q: @q
 %table{:border => "1"}
   %thead
     %tr
-      %th.article-title Title
+      %th.article-title
+        =sort_link(@q, :title, 'Title', default_order: :desc)
+      %th.article-author
+        =sort_link(@q, :user_email, 'Author')
       %th.article-text Content
       %th.article-action Action
   %tbody
+    - if @articles.empty?
+      %tr
+        %td{colspan: 4, style: 'text-align: center'} No articles to display
```

Comment:
I know this ain't for learning css and js but inline style are not preferred anywhere.